### PR TITLE
Adds env var and iam to allow fetching tags from lambda and log groups

### PR DESCRIPTION
**Pull request by srhopkins**
## what
* Adds the ability to fetch tags from lambda functions and log groups

## why
* To allow for better organization of metrics and logs in datadog dashboards, search and monitoring

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -99,4 +112,30 @@ data "aws_iam_policy_document" "lambda_default" {
...
+  dynamic "statement" {
+    for_each = var.dd_fetch_lambda_tags ? [true] : []
+    content {
+      sid    = "DdFetchLambdaTags"
+      effect = "Allow"
+
+      actions = [
+        "tag:GetResources"
+      ]
+      resources = ["arn:aws:lambda:::*"]
```
Comment (jamengual) [inline main.tf:125]: this is too open, this should be the lambda resource address created by this module

File: main.tf [hunk 2]
```diff
@@ -99,4 +112,30 @@ data "aws_iam_policy_document" "lambda_default" {
...
+  dynamic "statement" {
+    for_each = var.dd_fetch_log_group_tags ? [true] : []
+    content {
+      sid    = "DdFetchLogGroupTags"
+      effect = "Allow"
+
+      actions = [
+        "logs:ListTagsLogGroup"
+      ]
+      resources = ["logs:*"]
```
Comment (jamengual) [inline main.tf:138]: same here.

File: main.tf (hunk 1 above)
Comment (srhopkins) [reply main.tf:125]: It allows the lambda to fetch the tags from other lambdas and corrolate them to the cloudwatch log group. Right now the tags added to the customer's lambda functions do not get added.

File: main.tf (hunk 2 above)
Comment (srhopkins) [reply main.tf:138]: see my previous, similar scenerio

File: main.tf (hunk 1 above)
Comment (srhopkins) [reply main.tf:125]: See: https://docs.datadoghq.com/logs/guide/forwarder/#log-forwarding-optional

Comment (jamengual) [reply main.tf:125]: but why `arn:aws:lambda:::*` can't be the lambda arns passed on a list? by adding an new input variable?

**Pull request closed without merging**